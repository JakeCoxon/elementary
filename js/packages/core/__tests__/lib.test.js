import {
  el,
} from '..';


test('errors on graph construction', function() {
  expect(() => {
    // Missing second argument, should throw
    let p = el.seq({}, 1);
  }).toThrow();

  expect(() => {
    // Invalid node types
    let p = el.mul(1, 2, '4');
  }).toThrow();
});

test('clamp', function() {
  let p = el.clamp(-0.1, 0, 1);
  expect(p).toMatchSnapshot();

  let p2 = el.clamp({}, -0.1, 0, 1);
  expect(p2).toMatchSnapshot();
})

test('lerp', function() {
  let p = el.lerp(0.4, 0, 1);
  expect(p).toMatchSnapshot();

  let p2 = el.lerp({}, 0.4, 0, 1);
  expect(p2).toMatchSnapshot();
})