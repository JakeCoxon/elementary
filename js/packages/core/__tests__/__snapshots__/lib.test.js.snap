// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clamp 1`] = `
{
  "children": {
    "hd": {
      "children": 0,
      "hash": 276154007,
      "kind": "const",
      "outputChannel": 0,
      "props": {
        "value": -0.1,
      },
      "symbol": "__ELEM_NODE__",
    },
    "tl": {
      "hd": {
        "children": {
          "hd": {
            "children": 0,
            "hash": 745245079,
            "kind": "const",
            "outputChannel": 0,
            "props": {
              "value": 0,
            },
            "symbol": "__ELEM_NODE__",
          },
          "tl": {
            "hd": {
              "children": 0,
              "hash": 787097512,
              "kind": "const",
              "outputChannel": 0,
              "props": {
                "value": 1,
              },
              "symbol": "__ELEM_NODE__",
            },
            "tl": 0,
          },
        },
        "hash": 209048344,
        "kind": "max",
        "outputChannel": 0,
        "props": {},
        "symbol": "__ELEM_NODE__",
      },
      "tl": 0,
    },
  },
  "hash": 672731954,
  "kind": "min",
  "outputChannel": 0,
  "props": {},
  "symbol": "__ELEM_NODE__",
}
`;

exports[`clamp 2`] = `
{
  "children": {
    "hd": {
      "children": 0,
      "hash": 276154007,
      "kind": "const",
      "outputChannel": 0,
      "props": {
        "value": -0.1,
      },
      "symbol": "__ELEM_NODE__",
    },
    "tl": {
      "hd": {
        "children": {
          "hd": {
            "children": 0,
            "hash": 745245079,
            "kind": "const",
            "outputChannel": 0,
            "props": {
              "value": 0,
            },
            "symbol": "__ELEM_NODE__",
          },
          "tl": {
            "hd": {
              "children": 0,
              "hash": 787097512,
              "kind": "const",
              "outputChannel": 0,
              "props": {
                "value": 1,
              },
              "symbol": "__ELEM_NODE__",
            },
            "tl": 0,
          },
        },
        "hash": 209048344,
        "kind": "max",
        "outputChannel": 0,
        "props": {},
        "symbol": "__ELEM_NODE__",
      },
      "tl": 0,
    },
  },
  "hash": 672731954,
  "kind": "min",
  "outputChannel": 0,
  "props": {},
  "symbol": "__ELEM_NODE__",
}
`;

exports[`lerp 1`] = `
{
  "children": {
    "hd": {
      "children": {
        "hd": {
          "children": {
            "hd": {
              "children": 0,
              "hash": 787097512,
              "kind": "const",
              "outputChannel": 0,
              "props": {
                "value": 1,
              },
              "symbol": "__ELEM_NODE__",
            },
            "tl": {
              "hd": {
                "children": 0,
                "hash": 787097512,
                "kind": "const",
                "outputChannel": 0,
                "props": {
                  "value": 1,
                },
                "symbol": "__ELEM_NODE__",
              },
              "tl": 0,
            },
          },
          "hash": 1365355073,
          "kind": "sub",
          "outputChannel": 0,
          "props": {},
          "symbol": "__ELEM_NODE__",
        },
        "tl": {
          "hd": {
            "children": 0,
            "hash": 332596981,
            "kind": "const",
            "outputChannel": 0,
            "props": {
              "value": 0.4,
            },
            "symbol": "__ELEM_NODE__",
          },
          "tl": 0,
        },
      },
      "hash": 1389947663,
      "kind": "mul",
      "outputChannel": 0,
      "props": {},
      "symbol": "__ELEM_NODE__",
    },
    "tl": {
      "hd": {
        "children": {
          "hd": {
            "children": 0,
            "hash": 787097512,
            "kind": "const",
            "outputChannel": 0,
            "props": {
              "value": 1,
            },
            "symbol": "__ELEM_NODE__",
          },
          "tl": {
            "hd": {
              "children": 0,
              "hash": 745245079,
              "kind": "const",
              "outputChannel": 0,
              "props": {
                "value": 0,
              },
              "symbol": "__ELEM_NODE__",
            },
            "tl": 0,
          },
        },
        "hash": 207603630,
        "kind": "mul",
        "outputChannel": 0,
        "props": {},
        "symbol": "__ELEM_NODE__",
      },
      "tl": 0,
    },
  },
  "hash": 86843759,
  "kind": "add",
  "outputChannel": 0,
  "props": {},
  "symbol": "__ELEM_NODE__",
}
`;

exports[`lerp 2`] = `
{
  "children": {
    "hd": {
      "children": {
        "hd": {
          "children": {
            "hd": {
              "children": 0,
              "hash": 787097512,
              "kind": "const",
              "outputChannel": 0,
              "props": {
                "value": 1,
              },
              "symbol": "__ELEM_NODE__",
            },
            "tl": {
              "hd": {
                "children": 0,
                "hash": 787097512,
                "kind": "const",
                "outputChannel": 0,
                "props": {
                  "value": 1,
                },
                "symbol": "__ELEM_NODE__",
              },
              "tl": 0,
            },
          },
          "hash": 1365355073,
          "kind": "sub",
          "outputChannel": 0,
          "props": {},
          "symbol": "__ELEM_NODE__",
        },
        "tl": {
          "hd": {
            "children": 0,
            "hash": 332596981,
            "kind": "const",
            "outputChannel": 0,
            "props": {
              "value": 0.4,
            },
            "symbol": "__ELEM_NODE__",
          },
          "tl": 0,
        },
      },
      "hash": 1389947663,
      "kind": "mul",
      "outputChannel": 0,
      "props": {},
      "symbol": "__ELEM_NODE__",
    },
    "tl": {
      "hd": {
        "children": {
          "hd": {
            "children": 0,
            "hash": 787097512,
            "kind": "const",
            "outputChannel": 0,
            "props": {
              "value": 1,
            },
            "symbol": "__ELEM_NODE__",
          },
          "tl": {
            "hd": {
              "children": 0,
              "hash": 745245079,
              "kind": "const",
              "outputChannel": 0,
              "props": {
                "value": 0,
              },
              "symbol": "__ELEM_NODE__",
            },
            "tl": 0,
          },
        },
        "hash": 207603630,
        "kind": "mul",
        "outputChannel": 0,
        "props": {},
        "symbol": "__ELEM_NODE__",
      },
      "tl": 0,
    },
  },
  "hash": 86843759,
  "kind": "add",
  "outputChannel": 0,
  "props": {},
  "symbol": "__ELEM_NODE__",
}
`;
